packages/client/lib/rpc/modules/eth.ts

```diff
@@ -986,4 +989,37 @@ export class Eth {
     const block = await getBlockByOption(blockOpt, this._chain)
     return intToHex(block.transactions.length)
   }
+
+  async feeHistory(_params = []) {
+    const latest = await this._chain.getCanonicalHeadHeader()
+    return {
+      baseFeePerGas: [
+        '0x' + latest.baseFeePerGas?.toString(16),
+        '0x' + latest.baseFeePerGas?.toString(16),
+      ],
+      oldestBlock: '0x' + latest.number.toString(16),
+      gasUsedRatio: [0.5],
+    }
+  }
+
+  /**
+   * Returns an estimate of the current gas price based on the most recent block
+   * @returns the current gasPrice in gwei as a hex prefixed string
+   */
+  async gasPrice() {
+    let gasPrice: bigint = BigInt(0)
+    const latest = await this._chain.getCanonicalHeadHeader()
+    if (this._vm !== undefined && this._vm._common.isActivatedEIP(1559)) {
+      gasPrice = latest.calcNextBaseFee()
+    } else {
+      if (latest.gasUsed > BigInt(0)) gasPrice = latest.gasUsed
+      else {
+        while (gasPrice === BigInt(0)) {
+          const nextBlock = await this._chain.getBlock(latest.number - BigInt(1))
```

Comment:
are you looping always on the same block ?

Reply:
This is out of date. We merged a separate PR recently for `eth_gasPrice`.

Reply:
So to add: see comment here https://github.com/ethereumjs/ethereumjs-monorepo/pull/2413#issuecomment-1305497686